### PR TITLE
Shorten instance/<instanceID>/events/:name endpoint response

### PR DESCRIPTION
**Pull request by piotrek6641**
Make events data more readable with removal of redundant data in output
`http://0.0.0.0:8000/api/v1/instance/{instanceId}/events/{eventName}`

before:
```
{ "eventName": "msg", "message": { "a": 1 } }
{ "eventName": "msg", "message": { "a": 2 } }
{ "eventName": "msg", "message": { "a": 3 } }
```
now returns:
```
{ "a": 1 }
{ "a": 2 }
{ "a": 3 }
```
without message used to return:
```
{ "eventName": "msg"  }
{ "eventName": "msg"  }
{ "eventName": "msg"  }
```
now returns 
```
 {}
 {}
 {}
```

### Screenshots
events with no message:
image

events with object as a message:
image

events with string as a message:
image

**Clickup Task:** 
https://app.clickup.com/t/24308805/VDM-1283
**Issue:**
https://github.com/scramjetorg/transform-hub/issues/328

### CODE REVIEW

File: packages/types/src/api-expose.ts [hunk 1]
```diff
@@ -6,7 +6,10 @@ import { ControlMessageCode, MonitoringMessageCode } from "./message-streams";
 import { IObjectLogger } from "./object-logger";
 import { MaybePromise } from "./utils";
 
-export type ParsedMessage = IncomingMessage & { body?: any; params?: { [key: string]: any } | undefined };
+export type ParsedMessage = IncomingMessage & {
+    body?: any;
+    params?: { [key: string]: any },
+    query?: { [key: string]: any }};
```
Comment (patuwwy) [inline packages/types/src/api-expose.ts:12]:
```suggestion
    query?: { [key: string]: any }
};
```

**Review by ErykSol — APPROVED** [review]

File: packages/host/src/lib/csi-controller.ts [hunk 2]
```diff
@@ -672,7 +672,15 @@ export class CSIController extends TypedEmitter<Events> {
             }
 
             const out = new DataStream();
-            const handler = (data: any) => out.write(data);
+            const handler = (data: any) => {
+                if (typeof data !== "object") {
+                    out.write(data);
```
Comment (patuwwy) [inline packages/host/src/lib/csi-controller.ts:677]: missing return after write

**Review by patuwwy — CHANGES_REQUESTED** [review]

File: packages/host/src/lib/csi-controller.ts [hunk 3]
```diff
@@ -672,7 +672,15 @@ export class CSIController extends TypedEmitter<Events> {
...
+                }
+                // eslint-disable-next-line @typescript-eslint/no-unused-vars
+                const { _eventName, ...message } = data;
```
Comment (patuwwy) [inline packages/host/src/lib/csi-controller.ts:680]: there is no `_eventName` in event message payload so this assignment will cause `message` to be `data`.

File: packages/host/src/lib/csi-controller.ts [hunk 4]
```diff
@@ -672,7 +672,15 @@ export class CSIController extends TypedEmitter<Events> {
...
+
+                out.write(message ? message : {});
```
Comment (patuwwy) [inline packages/host/src/lib/csi-controller.ts:682]: pls return just message from data or undefined if message is empty

File: packages/host/src/lib/csi-controller.ts [hunk 5]
```diff
@@ -672,7 +672,15 @@ export class CSIController extends TypedEmitter<Events> {
...
+            };
```
Comment (patuwwy) [inline packages/host/src/lib/csi-controller.ts:683]:
```suggestion
            const handler = (data: any) => {
                if (typeof data !== "object") {
                    out.write(data);
                    return;
                }

                out.write(data.message);
            };
```

**Review by patuwwy — APPROVED** [review]

**Review by patuwwy — APPROVED** [review]

**Comment by a-tylenda** [comment]
I verified and endpoint `/api/v1/instance/{instanceId}/events/{eventName}` works as described in PR description, but...
there is a discrepancy between API and CLI. This is because under CLI command `si inst event on <id>` there is `getEvent()` method [this.router.get](https://github.com/scramjetorg/transform-hub/blob/feat/events-stream-format/packages/host/src/lib/csi-controller.ts#L713) not [this.router.upstream](https://github.com/scramjetorg/transform-hub/blob/feat/events-stream-format/packages/host/src/lib/csi-controller.ts#L668) edited by @piotrek6641

**Pull request merged**